androidapp/app/src/main/java/com/example/lanweather/data/entity/CurrentEntity.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.lanweather.data.entity
+
+import androidx.room.*
+import com.example.lanweather.data.model.Current
+
+@Entity(tableName = "Current")
+data class CurrentEntity (
+    @ColumnInfo(name = "temperature")
+    val temperature: Double?,
+    @ColumnInfo(name = "maxTemperature")
+    val maxTemperature: Double?,
+    @ColumnInfo(name = "minTemperature")
+    val minTemperature: Double?,
+    @ColumnInfo(name = "relativeHumidity")
+    val relativeHumidity: Int?,
+    @ColumnInfo(name = "windChill")
+    val windChill: Double?,
+    @ColumnInfo(name = "windDirection")
+    val windDirection: Double?,
+    @PrimaryKey
+    @ColumnInfo(name = "windSpeed")
+    val windSpeed: Double,
+    @ColumnInfo(name = "windGust")
+    val windGust: Double?,
+    @Embedded
+    val weather: WeatherEntity?,
+    @ColumnInfo(name = "probabilityOfPrecipitation")
+    val probabilityOfPrecipitation: Int?
+){
+    @Ignore
+    constructor(current: Current) : this (
```

Comment:
this isn't something i've encountered when using room or kotlin, what's its purpose?

Reply:
i think i get it but if you can explain it anyway for everyone else that'd be useful.

Reply:
The constructor allows the higher levels of json to create their lower members. No idea if this is the best way to handle this but it seems to do the job based on the resources I've been referencing.

Reply:
The constructor fills the database with initial data right? In this case it is a "Current" object that contains this info?

Reply:
Yeah. So based on the json, an nws object has a current object. So when we want to put an nws object in, part of that process is calling here to put in the current object field of the nws